### ISSUE TITLE
Class instance variables should not appear as members of class type

### ISSUE THREAD

**Issue by jakebailey** [microsoft/python-language-server#1233]
![image](https://user-images.githubusercontent.com/5341706/59868840-b5e22000-9346-11e9-95f7-649acf8e0500.png)

`Foo` does not have that member, `Foo()` does.

**Comment by MikhailArkhipov** [comment]
They are part of the type definition, so I guess should be filtered out by completion based on the class/static attributes.

**Comment by jakebailey** [comment]
It's not a completion thing, it's past that.

![image](https://user-images.githubusercontent.com/5341706/59886374-8564ab00-9373-11e9-8229-f25f4db8c000.png)

Note how `x` has `int`; it's not that we're listing the wrong members, that member shouldn't be on the type at all, just the instance.

**Comment by MikhailArkhipov** [comment]
Except our instances don't have ability to have members. Like in .NET types, all members are on the type but differ by binding flags. https://docs.microsoft.com/en-us/dotnet/api/system.type.getmethods?view=netframework-4.8#System_Type_GetMethods_System_Reflection_BindingFlags_

**Comment by MikhailArkhipov** [comment]
Functions have `IsClassMethod` and `IsStatic`, `IsOverload`. I think we may want to switch to flags, perhaps unify some with properties and then filter at completion time.

**Comment by jakebailey** [comment]
My only worry is that by doing filtering at completion time, we have to duplicate that code for member accesses in other places. We can filter the completion, but any code that (incorrectly) accesses that will be treated as valid. Maybe all members need those flags, not just functions/properties.

**Comment by MikhailArkhipov** [comment]
Yes, it may be the case. i.e. changing signature of `GetMemberNames` to include binding flags.
